src/core/thread/topology.hpp

```diff
@@ -129,16 +370,275 @@ class Child : public Neighbor
         kMaxRequestTlvs        = 5,
     };
 
+    /**
+     * This method clears the IPv6 addresses for the child.
+     *
+     */
+    void ClearIp6Addresses(void) { memset(mIp6Address, 0, sizeof(mIp6Address)); }
+
+    /**
+     * This method gets the IPv6 address at index @p aIndex.
+     *
+     * @param[in]  aIndex  The index into the IPv6 address list.
+     *
+     * @returns A reference to the IPv6 address entry at index @p aIndex.
+     *
+     */
+    Ip6::Address &GetIp6Address(uint8_t aIndex) { return mIp6Address[aIndex]; }
+
+    /**
+     * This method gets the child timeout.
+     *
+     * @returns The child timeout.
+     *
+     */
+    uint32_t GetTimeout(void) const { return mTimeout; }
+
+    /**
+     * This method sets the child timeout.
+     *
+     * @param[in]  aTimeout  The child timeout.
+     *
+     */
+    void SetTimeout(uint32_t aTimeout) { mTimeout = aTimeout; }
+
+    /**
+     * This method gets the network data version.
+     *
+     * @returns The network data version.
+     *
+     */
+    uint8_t GetNetworkDataVersion(void) const { return mNetworkDataVersion; }
+
+    /**
+     * This method sets the network data version.
+     *
+     * @param[in]  aVersion  The network data version.
+     *
+     */
+    void SetNetworkDataVersion(uint8_t aVersion) { mNetworkDataVersion = aVersion; }
+
+    /**
+     * This method generates a new challenge value to use during a child attach.
+     *
+     */
+    void GenerateChallenge(void) {
+        for (uint8_t i = 0; i < sizeof(mAttachChallenge); i++) {
+            mAttachChallenge[i] = static_cast<uint8_t>(otPlatRandomGet());
+        }
+    }
+
+    /**
+     * This method gets the current challenge value used during attach.
+     *
+     * @returns The current challenge value.
+     *
+     */
+    const uint8_t *GetChallenge(void) const { return mAttachChallenge; }
+
+    /**
+     * This method gets the challenge size (bytes) used during attach.
+     *
+     * @returns The challenge size (bytes).
+     *
+     */
+    uint8_t GetChallengeSize(void) const { return sizeof(mAttachChallenge); }
+
+    /**
+     * This method gets the IEEE 802.15.4 Frame Counter used during indirect retransmissions.
+     *
+     * @returns The IEEE 802.15.4 Frame Counter value.
+     *
+     */
+    uint32_t GetIndirectFrameCounter(void) const { return mIndirectSendInfo.mFrameCounter; }
```

Comment:
We can consider getting rid of `mIndirectSendInfo`struct as well... Mainly used this to group all indirect info together... (I can do it in a operate PR later)...